tests/hazmat/primitives/test_rsa.py

```diff
@@ -442,6 +444,73 @@ def test_pkcs1v15_signing(self, backend, subtests):
                 )
                 assert binascii.hexlify(signature) == example["signature"]
 
+    @pytest.mark.supported(
+        only_if=lambda backend: backend.rsa_padding_supported(
+            padding.PKCS1v15()
+        ),
+        skip_message="Does not support PKCS1v1.5.",
+    )
+    @pytest.mark.supported(
+        only_if=lambda backend: backend.signature_hash_supported(
+            hashes.SHA256()
+        ),
+        skip_message="Does not support SHA256 signature.",
+    )
+    def test_pkcs1v15_signing_without_digest(self, backend, subtests):
+        vectors = load_vectors_from_file(
+            os.path.join(
+                "asymmetric", "RSA", "FIPS_186-2", "SigVer15_186-3.rsp"
+            ),
+            load_rsa_nist_vectors,
+        )
+        for params in vectors:
+            with subtests.test():
+                # TODO Why those not match?
```

Comment:
This is failing because p and q are nominally a set (e.g., unordered) primes, but in practice many implementations require `p > q` so we changed how this recovery function returns (see the note in https://cryptography.io/en/latest/hazmat/primitives/asymmetric/rsa/#cryptography.hazmat.primitives.asymmetric.rsa.rsa_recover_prime_factors). Since you're asserting that the recovered `p == params["p"]` that fails when that value is contained in `params["q"]`